chore: add tests for explorer api abuse to watchdog canister

maksymar opened the pull request:
This PR adds a test case when API provider responds with a text string instead of a valid JSON.

ielashi commented:
I don't understand how the title and description of this PR relate to the actual change. The changes that I see are making adjustments to logs and adding tests. How does that fix how we handle API abuse?

maksymar commented:
> I don't understand how the title and description of this PR relate to the actual change. The changes that I see are making adjustments to logs and adding tests. How does that fix how we handle API abuse?

good catch, thank you! I must have been tired yesterday when debugging watchdog canister. switched error print messages to panics for debugging purposes and saw a reject message about abusing explorer api (also used 5s ping rate instead of a default 60s for debugging) and incorrectly concluded that getting a string instead of a proper JSON was causing the failure.

still I think it brings value to add such a test case for checking a correct handling of an unexpected text instead of a proper JSON. updated the PR title.
